lib/dynflow/action.rb

```diff
@@ -583,6 +583,12 @@ def root_action?
       @triggering_action.nil? && (@caller_action_id.nil? || @caller_execution_plan_id)
     end
 
+    # Plan an +event+ to be send to the action defined by +action+, what defaults to be self.
+    # if +time+ is not passed, event is sent as soon as possible.
+    def plan_event(event, time = nil, execution_plan_id: self.execution_plan_id, step_id: self.run_step_id)
+      world.executor.core.ask([:plan_event, Dispatcher::Event[execution_plan_id, step_id, event], time])
```

Comment:
When the worker and orchestrator run in  different process, then `world.executor` is `nil` here. How about making  `plan_event` store the timed events on the action and then processing (dispatching/sending to orchestrator) them when the work item is done?